### PR TITLE
[kueue] Add kjobctl job

### PR DESCRIPTION
**Pull request by trasc**
Add a dedicated job for kjobctl

### CODE REVIEW

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ presubmits:
     cluster: eks-prow-build-cluster
     branches:
     - ^main
-    skip_if_only_changed: "^docs/|^\\.github/|\\.(md)$|^(README|LICENSE|OWNERS)$"
+    skip_if_only_changed: "^docs/|^\\.github/|\\.(md)$|^(README|LICENSE|OWNERS)$|^(keps|cmd/experimental/kjobctl)/|^(site|charts)/"
```
[the remark this answers is not in the document]
Comment (trasc) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml]: done

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml [hunk 2]
```diff
@@ -252,7 +252,7 @@ presubmits:
     cluster: eks-prow-build-cluster
     branches:
     - ^main
-    skip_if_only_changed: "^docs/|^\\.github/|\\.(md)$|^(README|LICENSE|OWNERS)$"
+    skip_if_only_changed: "^docs/|^\\.github/|\\.(md)$|^(README|LICENSE|OWNERS)$|^(keps|cmd/experimental/kjobctl)/"
```
[the remark this answers is not in the document]
Comment (trasc) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml]: done

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml [hunk 3]
```diff
@@ -252,7 +252,7 @@ presubmits:
     cluster: eks-prow-build-cluster
     branches:
     - ^main
-    skip_if_only_changed: "^docs/|^\\.github/|\\.(md)$|^(README|LICENSE|OWNERS)$"
+    skip_if_only_changed: "^docs/|^\\.github/|^(README|LICENSE|OWNERS)$|^cmd/experimental/kjobctl/"
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml:255]: `site` can still be excluded.

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml [hunk 4]
```diff
@@ -345,3 +345,31 @@ presubmits:
...
+    annotations:
+      testgrid-dashboards: sig-scheduling
+      testgrid-tab-name: pull-kueue-kjobctl
+      description: "Run kjobctl tests"
+    spec:
+      containers:
+      - image: public.ecr.aws/docker/library/golang:1.22
+        command:
+        - make
+        args:
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml:364]: should we add verify here? for gofmt and other things.

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml (hunk 3 above)
Comment (trasc) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml:255]: No, in theory you could modify some generated files.

Comment (alculquicondor) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml:255]: gotcha, thanks

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml (hunk 4 above)
Comment (trasc) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml:364]: I think we could fine-tune this in the Makefile

**Pull request merged**